src/expr.rs

```diff
@@ -558,10 +573,18 @@ mod parse {
         Ok((input, Expr::Sqrt(Box::new(e))))
     }
 
+    fn h(input: &str) -> IResult<&str, Expr> {
+        let (input, e) = preceded(tag("H"), parenthised)(input)?;
+        Ok((input, Expr::H(Box::new(e))))
+    }
+
     fn atom(input: &str) -> IResult<&str, Expr> {
         let (input, sign) = opt(delimited(space0, tag("-"), space0))(input)?;
-        let (input, result) =
-            delimited(space0, alt((parenthised, exp, log, sqrt, lit, var)), space0)(input)?;
+        let (input, result) = delimited(
+            space0,
+            alt((parenthised, exp, log, sqrt, h, lit, var)),
```

Comment:
This should be no longer needed, _unless_ you want specific simplification rules for `H`.